readthedocs/rtd_tests/tests/test_yml_schema.py

```diff
@@ -320,6 +358,84 @@ def test_sphinx_invalid(tmpdir, value):
     )
 
 
+def test_sphinx_fail_on_warning(tmpdir):
+    content = '''
+version: "2"
+sphinx:
+  fail_on_warning: true
+    '''
+    assertValidConfig(tmpdir, content)
+
+
+@pytest.mark.parametrize('value', ['not true', "''", '[]'])
+def test_sphinx_fail_on_warning_invalid(tmpdir, value):
+    content = '''
+version: "2"
+sphinx:
+  fail_on_warning: {value}
+    '''
+    assertInvalidConfig(
+        tmpdir,
+        content.format(value=value),
+        ['is not a bool']
+    )
+
+
+def test_mkdocs(tmpdir):
+    content = '''
+version: "2"
+mkdocs:
+  configuration: mkdocs.yml
+    '''
+    assertValidConfig(tmpdir, content)
+
+
+def test_mkdocs_default_value(tmpdir):
+    content = '''
+version: "2"
+mkdocs:
+  file: mkdocs.yml  # Default value for configuration key
+    '''
+    assertValidConfig(tmpdir, content)
+
+
+@pytest.mark.parametrize('value', ['2', 'environment.py'])
```

Comment:
Why it does fail with `environment.py`? I think that it's a valid path and means that the `environment.py` file is at the root level of the repository.

(there are other places where I have the same question --for example, `conda.environment`)

Reply:
It fails because it doesn't exist in the current file system (there is only an `environment.yaml` file)

Reply:
I go a little crazy with the names on tests p:

I'm not sure how to validate a valid path without doing it in a hacky way or a complicated regex, not sure if it's worth it.